mesop/dataclass_utils/dataclass_utils.py

```diff
@@ -11,6 +12,7 @@
 
 _PANDAS_OBJECT_KEY = "__pandas.DataFrame__"
 _DIFF_ACTION_DATA_FRAME_CHANGED = "data_frame_changed"
+_DATETIME_OBJECT_KEY = "__mesop_datetime__"
```

Comment:
[nit] Can you move this line up below _PANDAS_OBJECT_KEY since those are related constants. I think for the name, maybe '__datetime.datetime__' since that's the module name.